docs/design/build-system.md

```diff
@@ -0,0 +1,154 @@
+# Build System
+
+Angular CLI includes a first-party build system for Angular applications distributed as `@angular-devkit/build-angular`.
+This build system is responsible for creating a standalone application from user source files and third-party dependencies.
+
+`@angular-devkit/build-angular` itself integrates with the rest of Angular CLI by being an [Architect builder](https://angular.io/guide/cli-builder).
+This document describes a top level view of the functionality in `@angular-devkit/build-angular`, referred as just the "build system".
+Deprecated or soon to be removed features are not described here.
+
+In broad strokes the main areas are:
+- loading and processing sources
+- code splitting
+- production optimizations
+- post-processing steps
+
+Many tools are used in this process, and most of these steps happen within a [Webpack](https://webpack.js.org/) build.
+We maintain a number of Webpack-centric plugins in this repository, some of these are public but most are private since they are very specific to our setup.
+
+
+## Loading and processing sources
+
+Sources for Angular CLI browser apps are comprised of TypeScript files, style sheets, assets, scripts, and third party dependencies.
+A given build will load these sources from disk, process them, and bundle them together.
+
+
+### TypeScript
```

Comment:
It might be good to explain what the AOT compilation actually does (and what is deferred when using JIT instead). My understanding is that it just compiles templates. In my also be good to discuss how the pipeline diverges/merges between the AOT and JIT forms.

Reply:
Added a link to https://angular.io/guide/aot-compiler. I'd prefer to not try and define that here if possible, as the build system mostly consumes.

I touched slightly on JIT on the optimization sections by saying `Large projects can also opt-out of AOT compilation for faster rebuilds.`, but that was all. By default we try to focus on AOT, which is now the default as well. Ideally we'd just do AOT even.

What things do you think belong in a JIT vs AOT section?

Reply:
I was mainly thinking about discussing how the build process changes between the two. If the only AOT step is compiling templates, then we could say we just skip that piece. The AOT doc you linked seems to discuss that pretty clearly, so I'm fine with a just including a link to that.

Reply:
Hmm now that I think about it some more, there are a number of different code transformations we do that differ between JIT and AOT. I don't mention those anywhere and really should.